### ISSUE TITLE
Yuzu fails to build (Arch Linux)

### ISSUE THREAD

**Issue by ImVersing** [yuzu-emu/yuzu#2597]
Hi, I was having an issue building Yuzu on my Arch machine and a few other people have replicated the same error as seen [here](https://pastebin.com/g9665LXg). Issue occurred both on the Nightly/Canary builds and persisted through multiple systems with different cmake versions. This was only fixed by editing line 3475 of /externals/fmt/include/fmt/format.h from `FMT_CONSTEXPR internal::udl_formatter<Char, CHARS...> ""operator_format() {` to `FMT_CONSTEXPR internal::udl_formatter<Char, CHARS...> operator_format() {` omitting the quotation marks.

**Comment by lioncash** [comment]
Hi. Until a new release of [fmt](https://github.com/fmtlib/fmt) is released, you will need to pass `-DFMT_USE_USER_DEFINED_LITERALS=0"` as part of your compilation flags on the command line.

**Comment by EchoDuplication** [comment]
Doesn't appear to work either @lioncash

**Comment by Shoegzer** [comment]
Having the same issue on the test system here, which is Linux Mint 19.1 x64 and kernel 4.18.  Tried with both GCC8 and GCC9 but compile still fails.  @lioncash the fmt fix doesn't work for me either (but thanks for trying to help).

**Comment by Shoegzer** [comment]
Update: tried compilation again with 6be79bab37a3ed8d9d1c194e85a112fff02d9150 but still failing.  Here's a relevant excerpt of the compile log:

```
                 from /home/shoegzer/Desktop/yuzu/yuzu-canary/externals/dynarmic/src/common/fp/fused.cpp:7:
/home/shoegzer/Desktop/yuzu/yuzu-canary/externals/fmt/include/fmt/format.h:3475:55: error: ISO C++ did not adopt string literal operator templates taking an argument pack of characters [-Wpedantic]
 3475 | FMT_CONSTEXPR internal::udl_formatter<Char, CHARS...> operator""_format() {
      |                                                       ^~~~~~~~
compilation terminated due to -Wfatal-errors.
externals/dynarmic/src/CMakeFiles/dynarmic.dir/build.make:134: recipe for target 'externals/dynarmic/src/CMakeFiles/dynarmic.dir/common/fp/fused.cpp.o' failed
```
This seems to be a major issue as I assume compilation is currently not possible on many linux systems.

**Comment by Shoegzer** [comment]
@lioncash to be more specific, here's the exception thrown after doing a cmake with the compilation flag added:
```
CMake Warning:
  Manually-specified variables were not used by the project:

    FMT_USE_USER_DEFINED_LITERALS
```
Is there any other workaround for this?

**Comment by magiruuvelvet** [comment]
Have you tried to compile with clang 8+. I have no issues building yuzu on Gentoo Linux.

I can imagine your compiler doesn't support modern C++ and is just too old.

**Comment by Shoegzer** [comment]
Thanks... I'm not using clang (it's a gcc thing), though cmake/compiler is relatively new and within yuzu requirements.  
```
$ cmake --version
cmake version 3.10.2

$ gcc --version
gcc (Ubuntu 9.1.0-2ubuntu2~18.04) 9.1.0
```

**Comment by 5schatten** [comment]
Try this https://github.com/citra-emu/citra/issues/4766#issuecomment-491904254 `CXXFLAGS+=" -DFMT_USE_USER_DEFINED_LITERALS=0"`

**Comment by Shoegzer** [comment]
@5schatten Thanks, and I tried that, but it doesn't work.  I get the same fmt-related compile errors as before.

**Comment by Shoegzer** [comment]
To be specific, here's what I used:

$ cmake -E env CXXFLAGS+=" -DFMT_USE_USER_DEFINED_LITERALS=0" cmake .. -DCMAKE_BUILD_TYPE=Release -DYUZU_USE_QT_WEB_ENGINE=ON

**Comment by OliPelz** [comment]
Hi all!
all the above hints did not work for me and I could not get past the error on Fedora 30.
What worked was to run the cmake command and let it fail.
Then manually edit the generated ```CMakeCache.txt``` file and goto line that reads:
```bash
CMAKE_CXX_FLAGS:STRING=
```
and change to
```bash
CMAKE_CXX_FLAGS:STRING=-DFMT_USE_USER_DEFINED_LITERALS=0
```
After that the error went away and compiling went through.

Best,
Oli

**Comment by Shoegzer** [comment]
@OliPelz that worked for me too, thanks.

@lioncash can something be done to permit passing the flag at commandline vs. the above process, at least until a proper fix is made?  Seems this might reflect a greater issue with the way the cmake process is working but I could be wrong.

**Comment by ReinUsesLisp** [comment]
Hello. Is this still present in master?

**Comment by lat9nq** [comment]
Appears to have been fixed with #3735. At the very least, building yuzu works fine here on Arch Linux.
